docs/resources/database_postgresql.md

```diff
@@ -0,0 +1,76 @@
+# Resource: <resource name> delphix_database_postgresql
+
+In Delphix, a databsae/source is required in the environment using which a dsource can be created.
```

Comment:
"In the Delphix Platform, a Database (or Source Config) identifies the environment's location from which a specific source dataset can be ingested from. dSources are then created from these Databases."

Largely pulled the updated description from the updated glossary: https://cd.delphix.com/docs/latest/glossary-of-major-delphix-concepts